plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/blobstore/DefaultS3OutputStream.java

```diff
@@ -107,12 +107,12 @@ public void flush(byte[] bytes, int off, int len, boolean closing) throws IOExce
     private void upload(byte[] bytes, int off, int len) throws IOException {
         try (ByteArrayInputStream is = new ByteArrayInputStream(bytes, off, len)) {
             int retry = 0;
-            while (retry <= getNumberOfRetries()) {
+            while (retry <= 0) {
```

Comment:
we should get rid of the retry logic here entirely, there is no need for the while loop when we aren't retrying.